xiaomi.cpp

```diff
@@ -1,6 +1,110 @@
+    \param zclFrame the actual ZCL frame which holds the Xiaomi/Lumi FCC0 cluster command or attribute
+ */
+void DeRestPluginPrivate::handleXiaomiLumiClusterIndication(const deCONZ::ApsDataIndication &ind, deCONZ::ZclFrame &zclFrame)
+{
+    if (zclFrame.isDefaultResponse())
+    {
+        return;
+    }
+
+    QDataStream stream(zclFrame.payload());
+    stream.setByteOrder(QDataStream::LittleEndian);
+
+    bool isReadAttr = false;
+    bool isReporting = false;
+    if (zclFrame.isProfileWideCommand() && zclFrame.commandId() == deCONZ::ZclReadAttributesResponseId)
+    {
+        isReadAttr = true;
+    }
+    if (zclFrame.isProfileWideCommand() && zclFrame.commandId() == deCONZ::ZclReportAttributesId)
+    {
+        isReporting = true;
+    }
+
+    // Read ZCL reporting and ZCL Read Attributes Response
+    if (isReadAttr || isReporting)
+    {
+        const NodeValue::UpdateType updateType = isReadAttr ? NodeValue::UpdateByZclRead : NodeValue::UpdateByZclReport;
+
+        // bool configUpdated = false;
+        // bool stateUpdated = false;
+
+        while (!stream.atEnd())
+        {
+            quint16 attrId;
+            quint8 attrTypeId;
+
+            stream >> attrId;
+            if (isReadAttr)
+            {
+                quint8 status;
+                stream >> status;  // Read Attribute Response status
+                if (status != deCONZ::ZclSuccessStatus)
+                {
+                    continue;
+                }
+            }
+            stream >> attrTypeId;
+
+            deCONZ::ZclAttribute attr(attrId, attrTypeId, QLatin1String(""), deCONZ::ZclRead, false);
+
+            if (!attr.readFromStream(stream))
+            {
+                continue;
+            }
+
+            ResourceItem *item = nullptr;
+
+            switch (attrId)
+            {
+            case XIAOMI_ATTRID_SPECIAL_REPORT:
+            {
+                handleZclAttributeReportIndicationXiaomiSpecial(ind, zclFrame);
+            }
+                break;
+
+            case XIAOMI_ATTRID_DEVICE_MODE:
+            {
+
+            }
+                break;
+
+            case XIAOMI_ATTRID_MOTION_SENSITIVITY:
+            {
+                Sensor *sensor = getSensorNodeForAddressEndpointAndCluster(ind.srcAddress(), ind.srcEndpoint(), XIAOMI_CLUSTER_ID );
+
+                if (sensor)
+                {
+                    quint8 sensitivity = attr.numericValue().u8;
+                    item = sensor->item(RConfigSensitivity);
+
+                    if (item && item->toNumber() != sensitivity)
+                    {
+                        item->setValue(sensitivity);
+                        enqueueEvent(Event(RSensors, RConfigSensitivity, sensor->id(), item));
+                        updateSensorEtag(&*sensor);
+                        sensor->setNeedSaveDatabase(true);
+                        queSaveDb(DB_SENSORS, DB_SHORT_SAVE_DELAY);
+                    }
```

Comment:
Perhaps this part can be simplified a bit, now that https://github.com/dresden-elektronik/deconz-rest-plugin/pull/5089 is merged?

Reply:
Sure, done.